components/Post/Post.tsx

```diff
@@ -1,260 +1,141 @@
-    const [shouldPlayAtEndOfSeek, setShouldPlayAtEndOfSeek] = useState<boolean>(false);
-    const [shouldPlay, setShouldPlay] = useState<boolean>(true);
-    const [isPlaybackAllowed, setIsPlaybackAllowed] = useState<boolean>(false);
-
-    // TODO: change to S3 files 
-    const TempSoundUrl = require('../../assets/audio/assets_audio_assets_audio_file_example_MP3_700KB.mp3')
+    const [isFocused, setIsFocused] = useState<boolean>(false)
 
-    
-    function onPlaybackStatusUpdate( playbackStatus: AVPlaybackStatus ) {
-        if (!playbackStatus.isLoaded) {
-            // Updating UI for the unloaded state
-            setIsPlaybackAllowed(false)
-            setSoundDuration(undefined)
-            setSoundPosition(undefined)
-
-            if (playbackStatus.error) {
-              console.log(`Encountered a fatal error during playback: ${playbackStatus.error}`);
-            }
-          } else {
-            // Updating UI for the loaded state
-        
-            setIsPlaying(playbackStatus.isPlaying)
-            setIsLoading(playbackStatus.isBuffering)
-        
-            if (playbackStatus.didJustFinish && !playbackStatus.isLooping) {
-              // The player has just finished playing
-              loadAndPlayPost()
-            }
-            // Set slider options
-            setSoundDuration(playbackStatus.durationMillis)
-            setSoundPosition(playbackStatus.positionMillis)
-            setIsPlaybackAllowed(true)
-          }
-    }
-
-    async function loadAndPlayPost() {
-        setIsLoading(true)
-        console.log('Load and play Sound')
-        console.log(post.audio)
-        const initialStatus = { 
-            shouldPlay: true,
-            progressUpdateIntervalMillis: 100
-        }
-        const { sound } = await Audio.Sound.createAsync(
-            {uri: post.audio},
-            initialStatus,
-            onPlaybackStatusUpdate
-        );
-        setSound(sound)
-        setIsLoading(false)
-    }
-
-    async function playPost() {
-        if (sound?._loaded) {
-            console.log('Playing Sound');
-            await sound.playAsync()
-        } else {
-            console.log('error playing post')
-        }
-    }
-
-    async function pausePost() {
-        if (sound?._loaded) {
-            console.log('Pausing Sound');
-            await sound.pauseAsync()
+    function handleOnPress() {
+        if (isFocused) {
+            console.log("Post pressed again, pausing sound")
+            setIsFocused(false)
+            setFocusedPostIndex(undefined)
         } else {
-            console.log('error pausing post')
+            console.log("Post pressed, loading and playing sound")
+            setIsFocused(true)
+            setFocusedPostIndex(index)
         }
     }
 
-    async function onPlayPausePressed() {
-        if (isPlaying) {
-            pausePost()
+    useEffect(() => {
+        if (focusedPostIndex == index) {
+            setIsFocused(true)
```

Comment:
Är tanken att välja vilken `Post` som är fokuserad genom att skicka in ett index till **alla** `Post` components och se om den matchar dess index?

Reply:
Där har du en poäng hahah Det har jag inte tänkt på. Kan vara en najs grej att ändra på i framtiden kanske?

Reply:
Ja det känns som en sak vi kan optimera sen ifall vi har tid. Tror inte det blir någon performance skillnad med vårat antal posts haha 🤝